semgrep.yml

```diff
@@ -59,19 +59,26 @@ rules:
   - id: not-using-our-pcre-wrappers
     patterns:
       - pattern-either:
-          - pattern: Pcre.regexp
-          - pattern: Pcre.pmatch
-          - pattern: Pcre.exec
-          - pattern: Pcre.exec_all
-          - pattern: Pcre.split
+          - pattern: $PCRE.regexp
+          - pattern: $PCRE.pmatch
+          - pattern: $PCRE.exec
+          - pattern: $PCRE.exec_all
+          - pattern: $PCRE.split
+      - metavariable-pattern:
+          metavariable: $PCRE
+          pattern-either:
+            - pattern: Pcre
+            - pattern: Pcre2
     message: >-
-      You should use one of the equivalent functions in Pcre_, which
-      automatically sets some flags and handles exceptions.
+      You should use one of the equivalent functions in Regex (or
+      Legacy_regex), which automatically sets some flags and handles
+      exceptions.
     languages: [ocaml]
     severity: ERROR
     paths:
       exclude:
-        - Pcre_.ml
+        - Regex.ml
+        - Legacy_regex.ml
```

Comment:
ok, great, we have this already. See my other comment about having the compiler report deprecation alerts (`[@@ocaml.deprecated "..."]`).

Reply:
Copied from my other comment:

If I use the compiler attribute, will we run into issues with many warnings from the current usage in generic mode? Or will it somehow only warn on net-new usage?

Reply:
Oh, I see, you're suggesting then we would just silence it in the generic mode modules.